packages/core/services/DatabaseService/index.ts

```diff
@@ -92,6 +94,70 @@ export default abstract class DatabaseService {
         this.dataSourceToAnnotationsMap.clear();
     }
 
+    protected async checkDataSourceForErrors(name: string): Promise<string[]> {
+        const tableColumnsQueryResult = await this.query(`\
+            SELECT COLUMN_NAME                \
+            FROM INFORMATION_SCHEMA.COLUMNS   \
+            WHERE TABLE_NAME = '${name}'      \
+        `);
+        const columnsOnTable = tableColumnsQueryResult.reduce(
+            (accum: Set<string>, row) => accum.add(row.column_name),
+            new Set<string>()
+        );
+
+        if (!columnsOnTable.has("File Path")) {
+            let error =
+                '\
+                "File Path" column is missing in the data source. \
+                Check the data source header row for a "File Path" (case-sensitive) column name and try again.';
+
+            // Attempt to find a column with a similar name to "File Path"
+            const columns = Array.from(columnsOnTable);
+            const filePathLikeColumn =
```

Comment:
What happens here if there are multiple hits?  eg, a column named 'pathfinder' and a column named 'file'?

Reply:
It'll point to `pathfinder` in that case. This is just to get a "best" guess at what they were trying to name "File Path"